src/components/budget/budget.service.ts

```diff
@@ -237,7 +236,9 @@ export class BudgetService {
 
     const changes = this.budgetRecordsRepo.getActualChanges(br, input);
     this.privileges.for(session, BudgetRecord, br).verifyChanges(changes);
-    this.privileges.for(session, Budget, budget).verifyCan('edit');
+    this.privileges
+      .for(session, Budget, budget)
+      .verifyCan('edit', 'universalTemplateFile');
```

Comment:
Why make this change?

Reply:
It breaks with a 403 without it.

Reply:
After removing the `this.privileges.for(session, Budget, budget).verifyCan('edit');`, the update goes through. It seems that the edit policy privilege might be messing things up ([talking about this here](https://github.com/SeedCompany/cord-api-v3/pull/3138/files#diff-2ce6dc91076b6679b82573512d72c4b924876b2bde1e363c03f9353d4eb1e782R34)).

Reply:
> In more detail, the `verifyCan` was expecting a `prop` after the action.

No. It can be either the object level action or an action on a specific property.
The specific property was incorrect, that's not what we were trying to confirm edit access too.



> Is the 403 happening now because of the changes that were made in the `ProjectManagerPolicy` or `ProjectService.currentBudget`?

Idk, both? They are correlated. The idea is to move the check into the policies.
The only tricky part is we need to check this on BudgetRecord, but it doesn't have a status field to check against. Hence the added Budget check here. But obv something wasn't right.

Reply:
Okay.  Andrew just put that property there to get it working; he knew it wasn't a final solution.

I get that we're moving the checks into the policies.  Question though, why was the `canEditFinalized` call removed?

Reply:
It was supposed be replaced by `verifyCan('edit')` and the policies would do it for us.

Reply:
Trying here #3139 
Seeing a failure with engagement create flag, but not budgets. Does the test suite not cover this?